src/System.Collections.NonGeneric/tests/CaseInsensitiveHashCodeProviderTests.cs

```diff
@@ -0,0 +1,187 @@
+                    Assert.Equal(expected, provider.GetHashCode(a) == provider.GetHashCode(b));
+                }
+                catch (CultureNotFoundException)
+                {
+                    continue;
+                }
+                finally
+                {
+                    CultureInfo.CurrentCulture = orig;
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("hello", "HELLO", true)]
+        [InlineData("hello", "hello", true)]
+        [InlineData("HELLO", "HELLO", true)]
+        [InlineData("hello", "goodbye", false)]
+        [InlineData(5, 5, true)]
+        [InlineData(10, 5, false)]
+        [InlineData(5, 10, false)]
+        public static void Ctor_CultureInfo_ChangeCurrentCulture_GetHashCodeCompare(object a, object b, bool expected)
+        {
+            var cultureNames = new string[]
+            {
+                "cs-CZ","da-DK","de-DE","el-GR","en-US",
+                "es-ES","fi-FI","fr-FR","hu-HU","it-IT",
+                "ja-JP","ko-KR","nb-NO","nl-NL","pl-PL",
+                "pt-BR","pt-PT","ru-RU","sv-SE","tr-TR",
+                "zh-CN","zh-HK","zh-TW"
+            };
+
+            foreach (string cultureName in cultureNames)
+            {
+                try
+                {
+                    var provider = new CaseInsensitiveHashCodeProvider(new CultureInfo(cultureName));
+                    Assert.Equal(provider.GetHashCode(a), provider.GetHashCode(a));
+                    Assert.Equal(provider.GetHashCode(b), provider.GetHashCode(b));
+                    Assert.Equal(expected, provider.GetHashCode(a) == provider.GetHashCode(b));
+                }
+                catch (CultureNotFoundException)
+                {
+                    continue;
+                }
+            }
+        }
+
+        [Fact]
+        public static void Ctor_CultureInfo_GetHashCodeCompare_TurkishI()
+        {
+            var cultureNames = new string[]
+            {
+                "cs-CZ","da-DK","de-DE","el-GR","en-US",
+                "es-ES","fi-FI","fr-FR","hu-HU","it-IT",
+                "ja-JP","ko-KR","nb-NO","nl-NL","pl-PL",
+                "pt-BR","pt-PT","ru-RU","sv-SE","tr-TR",
+                "zh-CN","zh-HK","zh-TW"
+            };
+
+            foreach (string cultureName in cultureNames)
+            {
+                try
+                {
+                    var culture = new CultureInfo(cultureName);
+                    var provider = new CaseInsensitiveHashCodeProvider(culture);
+
+                    // Turkish has lower-case and upper-case version of the dotted "i", so the upper case of "i" (U+0069) isn't "I" (U+0049)
+                    // but rather "İ" (U+0130)
+                    Assert.Equal(
+                        culture.Name != "tr-TR",
+                        provider.GetHashCode("file") == provider.GetHashCode("FILE"));
+                }
+                catch (CultureNotFoundException)
```

Comment:
Does it make sense to move the catch of the CultureNotFoundException so it just guards the call to new `CultureInfo(cultureName)` so we would get failures if either the GetHashCode or CaseInsensitiveHashCodeProvider calls threw?